woodwork/table_accessor.py

```diff
@@ -38,24 +38,60 @@
 dd = import_or_none('dask.dataframe')
 ks = import_or_none('databricks.koalas')
 
+ColumnName = Hashable
+
 
 class WoodworkTableAccessor:
     def __init__(self, dataframe):
         self._dataframe_weakref = weakref.ref(dataframe)
         self._schema = None
 
-    def init(self,
-             index=None,
-             time_index=None,
-             logical_types=None,
-             already_sorted=False,
-             schema=None,
-             validate=True,
-             use_standard_tags=True,
-             **kwargs):
-        """Initializes Woodwork typing information for a DataFrame.
+    def init(self, schema=None, **kwargs):
```

Comment:
In general, we're adding a lot of new functionality with woodwork init here, and I think it could really benefit from an in-depth explanation somewhere. Maybe it makes sense to create an issue to make a guide on the different ways to initialize woodwork.